iml-agent/src/daemon_plugins/daemon_plugin.rs

```diff
@@ -27,8 +23,8 @@ pub fn as_output(x: impl serde::Serialize + Send) -> Result<Output> {
 /// to the `plugin_registry` below.
 pub trait DaemonPlugin: std::fmt::Debug {
     /// Returns full listing of information upon session esablishment
-    fn start_session(&mut self) -> Box<dyn Future<Item = Output, Error = ImlAgentError> + Send> {
-        Box::new(future::ok(None))
+    fn start_session(&mut self) -> Pin<Box<dyn Future<Output = Result<Output>> + Send>> {
```

Comment:
I'm not sure I understand the purpose of returning `Pin` in these functions. Just wondering if i'm missing an important concept here. My understanding of pinning is to "lock" a future to a particular spot in memory, allowing you to create references to values inside an async block without worrying about the reference moving (https://rust-lang.github.io/async-book/04_pinning/01_chapter.html). But seeing the use of Pin here makes me wonder if I understand its use correctly. What does it do in this case?

Reply:
Using [`boxed`](https://rust-lang-nursery.github.io/futures-api-docs/0.3.0-alpha.19/futures/future/trait.FutureExt.html#method.boxed) will return that type. 

I'd much rather these be async methods, but that is not supported in the initial async/await for 1.39.